packages/expo-modules-core/ios/Fabric/ExpoFabricView.swift

```diff
@@ -1,50 +1,64 @@
 // Copyright 2022-present 650 Industries. All rights reserved.
 
 @objc(ExpoFabricView)
-public class ExpoFabricView: ExpoFabricViewObjC {
+open class ExpoFabricView: ExpoFabricViewObjC, AnyExpoView {
   /**
    A weak reference to the app context associated with this view.
-   The app context is injected into the class when making its copy in runtime,
+   The app context is injected into the class after the context is initialized.
    see the `makeClass` static function.
    */
-  weak var appContext: AppContext? { __injectedAppContext() }
+  public weak var appContext: AppContext? { ExpoFabricView.appContextFromClass() }
 
   /**
-   Name of the module associated with this view. Injected by `makeClass` static function.
+   The view definition that setup from `ExpoFabricView.create()`.
    */
-  lazy var moduleName: String = __injectedModuleName()
-
-  /**
-   Returns a holder of the module associated with this view.
-   */
-  private var moduleHolder: ModuleHolder? {
-    return appContext?.moduleRegistry.get(moduleHolderForName: moduleName)
-  }
+  private var viewDefinition: AnyViewDefinition?
 
   /**
    A dictionary of prop objects that contain prop setters.
    */
-  lazy var viewManagerPropDict: [String: AnyViewProp]? = moduleHolder?.definition.view?.propsDict()
+  lazy var viewManagerPropDict: [String: AnyViewProp]? = viewDefinition?.propsDict()
 
   // MARK: - Initializers
 
-  /**
-   The default initializer for all native views. It is called by Fabric.
-   */
+  // swiftlint:disable unavailable_function
   @objc
   public init() {
+    // For derived views, their initializer should be replaced by the 'class_replaceMethod'.
+    fatalError("Unsupported direct init() call for ExpoFabricView.")
+  }
+  // swiftlint:enable unavailable_function
+
+  @objc
+  public override init(frame: CGRect) {
+    super.init(frame: frame)
+  }
+
+  required public init(appContext: AppContext? = nil) {
     super.init(frame: .zero)
-    initializeContentView()
   }
 
-  required init?(coder: NSCoder) {
+  /**
+   The view creator expected to be called for derived ExpoFabricView, the `viewDefinition` and event dispatchers will be setup from here.
+   */
+  internal static func create(viewType: ExpoFabricView.Type, viewDefinition: AnyViewDefinition, appContext: AppContext) -> ExpoFabricView {
+    let view = viewType.init(appContext: appContext)
+    view.viewDefinition = viewDefinition
+    assert(appContext == view.appContext)
+    view.installEventDispatchers()
+    return view
+  }
```

Comment:
> since we swizzle ViewManagerAdapter_ExpoImage.new() to ImageView.init(appContext:) and we need moduleHolder/viewDefintion in the contructor (for installEventDispatchers()). it doesn't give us chance to inject iMethod or iVar to the ImageView class for the moduleName. this pr tries to add dedicated ExpoFabricView.create() and passing viewDefinition into the class. not a perfect case but works though.

here is the place i'm not happy with. the call flow here is
`ViewManagerAdapter_ExpoImage.new()` -> `ViewDefinition.createView()` -> `ExpoFabricView.create()` -> `ImageView.init(appContext:)` -> `ExpoFabricView.init(appContext:)` -> set `viewDefinition` here.

Reply:
Yeah, I agree this path doesn't look great. Let's add a comment in the code. We could try to address this later in SDK 52.